src/components/CreateJobPostForm/CreateJobPostForm.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { ChangeEventHandler, useEffect, useState } from "react";
+import { Button } from "@mui/material";
+import LoadingButton from "@mui/lab/LoadingButton";
+import SaveIcon from "@mui/icons-material/Save";
+import { toast } from "react-toastify";
+import {
+  Control,
+  FieldErrors,
+  useForm,
+  UseFormSetValue,
+} from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import propsDataCollection from "components/CreateJobPostForm/staticData";
+import LayoutElementWithTitle from "components/layoutElementWithTitle/LayoutElementWithTitle";
+import { IJobPostFormFields } from "components/CreateJobPostForm/typesDef";
+import FileBox from "components/FileBox/FileBox";
+import { IFreelancer } from "services/profileInfo/typesDef";
+import { useGetOwnUserQuery } from "services/user/setUserAPI";
+import {
+  usePostJobPostMutation,
+  usePostDraftMutation,
+} from "services/jobPosts/setJobPostsAPI";
+import {
+  createJobPostValidationSchema,
+  jobPostsDraftSchema,
+} from "validation/createJobPostValidationSchema";
+import createDataForResolver from "utils/functions/createDataForResolver";
+import { SButtonsGroup, SWrapper } from "components/CreateJobPostForm/styles";
+import { createJobAttachmentFileTypes } from "utils/consts/fileTypes";
+
+function CreateJobPostForm() {
+  const { data: userData, isError: getUserError } = useGetOwnUserQuery();
+
+  const [
+    postJobPost,
+    {
+      isSuccess: postJobSuccess,
+      isError: postJobError,
+      isLoading: postJobLoading,
+    },
+  ] = usePostJobPostMutation();
+  const [
+    postDraft,
+    {
+      isSuccess: postDraftSuccess,
+      isError: postDraftError,
+      isLoading: postDraftLoading,
+    },
+  ] = usePostDraftMutation();
+  const [selectedFile, setSelectedFile] = useState<null | File>(null);
+  const [isDraft, setIsDraft] = useState(true);
```

Comment:
types. in this case its obvious, but lets follow same code style

Reply:
fixed